clang/include/clang/AST/Stmt.h

```diff
@@ -3302,21 +3302,20 @@ class GCCAsmStmt : public AsmStmt {
   friend class ASTStmtReader;
 
   SourceLocation RParenLoc;
-  StringLiteral *AsmStr;
+  Expr *AsmStr;
 
   // FIXME: If we wanted to, we could allocate all of these in one big array.
-  StringLiteral **Constraints = nullptr;
-  StringLiteral **Clobbers = nullptr;
+  Expr **Constraints = nullptr;
```

Comment:
Another suggestion to add to the `lets make these sane` bug report, we could probably use `ArrayRef` for these to make the pointer-pointer less difficult to use.